.github/workflows/checks.yml

```
@@ -1,8 +1,6 @@
 name: Run XO and Prettier
 on:
-  push:
-    branches:
-      - dev
+  pull_request:
 
```

I like jobs having the option of being triggered manually, again little downside and you never know when you might need it

```suggestion
  workflow_dispatch:
```